source/common/config/well_known_names.cc

```diff
@@ -0,0 +1,109 @@
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Config {
+
+std::vector<std::pair<std::string, std::string>> TagNameValues::getRegexMapping() {
+  std::vector<std::pair<std::string, std::string>> name_regex_pairs;
+
+  // Note: the default regexes are defined below in the order that they will typically be matched
+  // (see the TagExtractor class definition for an explanation of the iterative matching process).
+  // This ordering is roughly from most specific to least specific. Despite the fact that these
+  // regexes are defined with a particular ordering in mind, users can customize the ordering of the
+  // processing of the default tag extraction regexes and include custom tags with regexes via the
+  // bootstrap configuration. Because of this flexibility, these regexes are designed to not
+  // interfere with one another no matter the ordering. They are tested in forward and reverse
+  // ordering to ensure they will be safe in most ordering configurations.
+
+  // To give a more user-friendly explanation of the intended behavior of each regex, each is
+  // preceded by a comment with a simplified notation to explain what the regex is designed to
+  // match:
+  // - The text that the regex is intended to capture will be enclosed in ().
+  // - Other default tags that are expected to exist in the name (and may or may not have been
+  // removed before this regex has been applied) are enclosed in [].
+  // - Stand-ins for a variable segment of the name (including inside capture groups) will be
+  // enclosed in <>.
+  // - Typical * notation will be used to denote an arbitrary set of characters.
+
+  // *_rq(_<response_code>
+  name_regex_pairs.push_back({RESPONSE_CODE, "_rq(_(\\d{3}))$"});
+
+  // *_rq(_<response_code_class>xx)
+  name_regex_pairs.push_back({RESPONSE_CODE_CLASS, "_rq(_(\\dxx))$"});
+
+  // http.<stat_prefix>.dynamodb.table.<table_name>.capacity.<operation_name>.__partition_id=<last_seven_characters_from_partition_id>
+  name_regex_pairs.push_back({DYNAMO_PARTITION_ID, "^http(?=\\.).*?\\.dynamodb\\.table(?=\\.).*?\\."
+                                                   "capacity(?=\\.).*?(\\.__partition_id=(\\w{7}))"
+                                                   "$"});
+
+  // http.[<stat_prefix>.]dynamodb.operation.(<operation_name>.)<base_stat> or
+  // http.[<stat_prefix>.]dynamodb.table.[<table_name>.]capacity.(<operation_name>.)[<partition_id>]
+  name_regex_pairs.push_back({DYNAMO_OPERATION, "^http(?=\\.).*?\\.dynamodb.(?:operation|table(?="
+                                                "\\.).*?\\.capacity)(\\.(.*?))(?:\\.|$)"});
```

Comment:
Might be clearer as two regexes..

Reply:
As it works now, we use names as unique identifiers (and we throw if the user attempts to use duplicate names), so if a tag shows up in two distinct places a single regex must capture both.  This is how we currently ensure a default regex isn't used twice (one custom and one with the with the use_all_defaults flag for example).  One alternative would be to ensure uniqueness of the actual regex strings themselves during creation instead of names.  That would allow us to have duplicate names in the list of tag extractors.  It wouldn't be a big change; do you think that would be preferred?  My only hesitation is that it seems a little less intuitive for the user than ensuring uniqueness of names.

Reply:
Got it, it's good as is.